types/node/index.d.ts

```diff
@@ -5017,6 +5017,7 @@ declare module "stream" {
             unshift(chunk: any): void;
             wrap(oldStream: NodeJS.ReadableStream): Readable;
             push(chunk: any, encoding?: string): boolean;
+            _destroy(err: Error, callback: Function): void;
```

Comment:
I think this should be `_destroy(err: Error | null, (err: Error | null) => void)`.

Reply:
At least the documentation does not specify `null` as a possible value for the first parameter.

For the callback type I tried to align this change with the signature of `_destroy` in other stream classes. If it was changed to the more specific type it should probably be reflected there as well for consistency.

Reply:
To my understanding `_destroy()` gets called if `destroy()` is called. As parameter to `destroy()` is optional it may happen that `_destroy()` gets no Error object.
Node defaults to null in this case. see https://github.com/nodejs/node/blob/1fe9b535a78a23a9549b544ec4a4f76614f45c5c/lib/internal/streams/destroy.js#L32

But you are right, the `_destroy()` for `Writeable` and `Duplex` are like your proposal so maybe it's better to keep them in sync to avoid confusion.